src/helpers/BasicPointsAndBadges/ConnectExternalAccountActivity.ts

```diff
@@ -1,18 +1,26 @@
 import MyDataHelps from "@careevolution/mydatahelps-js";
 import type { BasicPointsForBadgesActivity, BasicPointsForBadgesActivityState } from "./Activities";
 
+export type ConnectExternalActivityType = "Provider" | "Health Plan" | "Device Manufacturer";
+
 export interface ConnectExternalAccountActivity extends BasicPointsForBadgesActivity {
     type: "connectExternalAccount";
-    providerCategories: string[];
+    providerCategories: ConnectExternalActivityType[];
 }
 
 interface ConnectExternalAccountActivityState extends BasicPointsForBadgesActivityState {
     providersConnected?: number[];
 }
 
 export async function awardConnectExternalAccountActivityPoints(activity: ConnectExternalAccountActivity, activityState: ConnectExternalAccountActivityState) {
+    //convert to array to protect against new types added in consumers but not yet here
+    const activityProviderCategories = activity.providerCategories as string[]; 
+
+    //Get the external accounts
     const connectedAccounts = await MyDataHelps.getExternalAccounts();
-    const allProviders = connectedAccounts.filter(account => activity.providerCategories.includes(account.provider.category)).map(t => t.id);
```

Comment:
As we discussed, I believe this should be using the `t.provider.id` instead of `t.id` since the external account connection ids can change if disconnected and reconnected.  The provider ids should remain the same.  Probably worth renaming `t` to `account` as well, for better understanding of what is being mapped at a glance.

Also, as discussed, the state for this activity should track all provider ids that contributed to the points, not just the current list of connected ones that still match the criteria.  Therefore, the `providersConnected` array should be being  appended to rather than replaced when new provider connections are found.